engine/crates/integration-tests/tests/federation/entity_caching/directive_scopes.rs

```diff
@@ -0,0 +1,187 @@
+//! Tests that we handle `@authenticated` & `@requiresScopes` directives on parent fields/types
+//! correctly when doing entity caching
+
+use engine_v2::Engine;
+use graphql_mocks::{FederatedInventorySchema, FederatedProductsSchema, FederatedReviewsSchema, SecureFederatedSchema};
+use integration_tests::{
+    federation::{EngineV2Ext, TestEngineV2},
+    openid::{CoreClientExt, OryHydraOpenIDProvider, JWKS_URI},
+    runtime,
+};
+
+#[test]
+fn test_authenticated_field_does_not_share_cache_with_unauthenticated() {
+    runtime().block_on(async move {
+        let token = jwt_token("read").await;
+
+        let engine = engine().await;
+
+        const AUTHENTICATED_QUERY: &str = "{ authenticatedProducts { upc reviews { id body } } }";
+        const UNAUTHENTICATED_QUERY: &str = "{ topProducts { upc reviews { id body } } }";
+
+        engine
+            .post(AUTHENTICATED_QUERY)
+            .header("Authorization", format!("Bearer {token}"))
+            .await
+            .into_data();
```

Comment:
why `into_data()` though?

Reply:
Why not into_data?

Reply:
I'm just confused as to why you're calling this method but not doing anything with the result

Reply:
Because it has an `assert!(errors == None)` in it

Reply:
might be worth adding a method that only does that explicitly then imho. Were I to touch that code I might remove `into_data()` not understanding why it's there to be honest, no idea.
I tend to always `insta` snapshot the result to ensure I did receive what I expected.

Reply:
Maybe don't remove things you don't understand?

Reply:
I'm really not _that_ keen on adding another method that does basically the same thing as into_data

Reply:
> Maybe don't remove things you don't understand?

Well yeah, it's a bit extreme what I said. It's more the opposite, someone can change `into_data()` to not have the side-effect anymore. So I'm not a fan of relying on a side-effect of a method with a name that has no relation to the side-effect you expect.

Reply:
Why would someone remove an assert from commonly used test code?

Reply:
🤷. My reasoning is just that the fact that I had to ask why you did it or that I would have to look at the definition to understand why you're doing this is reason enough for me to rename it to something more explicit. If you think that's useless then just leave it as is 🤷

Reply:
fair enough, I just use insta snapshots everywhere. Sometimes a bit tedious, but I find problems to be easy to spot and understand without having to touch the code.

Reply:
Yeah, I normally would use insta.  Just for these tests I don't really give a shit about the contents of the response, I just want to make sure the subgraph was called the correct number of times.